src/Product/Product.php

```diff
@@ -208,20 +208,105 @@ public function getPriceFrom($type = 'retail', $currencyID = null)
+	 * @param string $type                          The type of price, defaults to 'retail'
+	 * @param null|string $currencyID               The currency ID, will be the default currency if set to null
+	 * @param bool $returnHighest                   Set as true to return the highest price, and false to return the
+	 *                                              lowest price
+	 * @throws Exception\PriceNotFoundException     Throws exception if no price can be found
+	 *
+	 * @return string
+	 */
+	public function getOptionPrice(array $options, $type = 'retail', $currencyID = null, $returnHighest = true)
+	{
+		$prices = $this->getOptionPrices($options, $type, $currencyID);
+		sort($prices);
+
+		$price = ($returnHighest) ? array_pop($prices) : array_shift($prices);
+
+		if (false === $price) {
+			$optionsString = implode(', ', $options);
+			throw new Exception\PriceNotFoundException('Could not find price for product ' . $this->id . ', with options ' . $optionsString . ' and a currency ID of ' . $currencyID);
+		}
+
+		return $price;
+	}
+
+	/**
+	 * Get the lowest price for a specific option, acts as a shorthand alias for `getOptionPrice()` with $returnHighest
+	 * set to false
+	 *
+	 * @param array $options             The options to return the price for
+	 * @param string $type               The type of price, defaults to 'retail'
+	 * @param null|string $currencyID    The currency ID, will be the default currency if set to null
+	 *
+	 * @return string
+	 */
+	public function getOptionPriceFrom(array $options, $type = 'retail', $currencyID = null)
+	{
+		return $this->getOptionPrice($options, $type, $currencyID, false);
+	}
+
+	/**
+	 * Get the highest price for a specific option, acts as a shorthand alias for `getOptionPrice()` with $returnHighest
+	 * set to true
+	 *
+	 * @param array $options                The options to return the price for
+	 * @param string $type                  The type of price, defaults to 'retail'
+	 * @param null | string $currencyID     The currency ID, will be the default currency if set to null
+	 *
+	 * @return string
+	 */
+	public function getOptionPriceTo(array $options, $type = 'retail', $currencyID = null)
+	{
+		return $this->getOptionPrice($options, $type, $currencyID, true);
+	}
+
+	/**
+	 * Get an array of all unique prices for a specific option
+	 *
+	 * @param array $options             The options to return the price for
+	 * @param string $type               The type of price, defaults to 'retail'
+	 * @param null|string $currencyID    The currency ID, will be the default currency if set to null
+	 *
+	 * @return array
+	 */
+	public function getOptionPrices(array $options, $type = 'retail', $currencyID = null)
+	{
+		$currencyID = $currencyID ?: $this->_defaultCurrency;
+
+		$prices = [$this->getPrice($type, $currencyID)];
+
+		foreach ($this->getVisibleUnits() as $unit) {
+			foreach ($options as $name => $value) {
```

Comment:
Is this how we want to do it? Consider ['colour' => 'black', 'size' => 'L']. It will return All prices for black products and L products instead of black L products.